Solutions/Azure Active Directory/Analytic Rules/Brute Force Attack against GitHub Account.yaml

```diff
@@ -23,35 +23,31 @@ query: |
   let LearningPeriod = 7d; 
   let BinTime = 1h; 
   let RunTime = 1h; 
-  let StartTime = 1h; 
-  let NumberOfStds = 3; 
-  let MinThreshold = 10.0; 
+  let StartTime = 1h;  
+  let sensitivity = 2.5;
   let EndRunTime = StartTime - RunTime; 
   let EndLearningTime = StartTime + LearningPeriod;
   let aadFunc = (tableName:string){
-  let GitHubFailedSSOLogins = (table(tableName)  
-  | where AppDisplayName == "GitHub.com" 
-  | where ResultType != 0); 
-  GitHubFailedSSOLogins 
+  table(tableName)  
   | where TimeGenerated between (ago(EndLearningTime) .. ago(StartTime)) 
-  | summarize FailedLoginsCountInBinTime = count() by  UserPrincipalName, bin(TimeGenerated, BinTime), Type
-  | summarize AvgOfFailedLoginsInLearning = avg(FailedLoginsCountInBinTime), StdOfFailedLoginsInLearning = stdev(FailedLoginsCountInBinTime) by UserPrincipalName, Type
-  | extend LearningThreshold = max_of(AvgOfFailedLoginsInLearning + StdOfFailedLoginsInLearning * NumberOfStds, MinThreshold)  
-  | join kind=innerunique ( 
-    GitHubFailedSSOLogins 
-    | where TimeGenerated between (ago(StartTime) .. ago(EndRunTime)) 
-    | summarize FailedLoginsCountInRunTime = count() by User = Identity, UserPrincipalName, bin(TimeGenerated, BinTime), Type
-  ) on UserPrincipalName 
-  | where FailedLoginsCountInRunTime > LearningThreshold
-  | extend AccountCustomEntity = UserPrincipalName , timestamp = TimeGenerated
+  | where AppDisplayName =~ "GitHub.com"
+  | where ResultType != 0
+  | make-series FailedLogins = count() on TimeGenerated from ago(LearningPeriod) to ago(EndRunTime) step BinTime by UserPrincipalName, Type
+  | extend (Anomalies, Score, Baseline) = series_decompose_anomalies(FailedLogins, sensitivity, -1, 'linefit')
+  | mv-expand FailedLogins to typeof(double), TimeGenerated to typeof(datetime), Anomalies to typeof(double), Score to typeof(double), Baseline to typeof(long)  
+  | where TimeGenerated >= ago(RunTime)
+  | where Anomalies > 0
```

Comment:
Its a good idea to join back to raw events on the time that the anomalies occur. The anomalies themselves are hard to triage if you don't have the context of what the raw results that triggered the anomaly are.